pkg/feature/servicemesh/cleanup.go

```diff
@@ -0,0 +1,66 @@
+package servicemesh
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrlLog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/gvr"
+)
+
+var log = ctrlLog.Log.WithName("features")
+
+func RemoveExtensionProvider(f *feature.Feature) error {
+	ossmAuthzProvider := fmt.Sprintf("%s-odh-auth-provider", f.Spec.AppNamespace)
+
+	mesh := f.Spec.ControlPlane
+
+	smcp, err := f.DynamicClient.Resource(gvr.SMCP).
+		Namespace(mesh.Namespace).
+		Get(context.TODO(), mesh.Name, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Since the configuration of the extension provider is a patch, it could happen that
+			// the SMCP is already gone, and there will be nothing to unpatch.
+			return nil
+		}
+		return err
+	}
```

Comment:
Good catch! So I will pay back for fixing my mistake with a little simplification :)

```suggestion
	if err != nil {
		// Since the configuration of the extension provider is a patch, it could happen that
		// the SMCP is already gone, and there will be nothing to unpatch.
		return client.IgnoreNotFound(err)
	}
```